cdap-master/src/main/java/co/cask/cdap/data/runtime/main/TransactionServiceTwillRunnable.java

```diff
@@ -111,10 +95,11 @@ protected void doInit(TwillContext context) {
 
   @Override
   public void getServices(List<? super Service> services) {
-    services.add(zkClient);
-    services.add(kafkaClient);
-    services.add(metricsCollectionService);
-    services.add(txService);
+    services.add(injector.getInstance(ZKClientService.class));
+    services.add(injector.getInstance(KafkaClientService.class));
+    services.add(injector.getInstance(AuthorizationEnforcementService.class));
```

Comment:
The problem is, it's part of the injector. This mean it is needed by some class, hence it's safer to always start it.